### PR TITLE
Add referral form docs

### PR DESCRIPTION
**Pull request by paulodeon**
### Context

To make it easier for developers to get started 

### Changes proposed in this pull request

Add documentation highlighting the form structure 

### Guidance to review

Does it make sense?
Anything else we should add?

### Link to Trello card

No trello card

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

File: docs/referrer-form-structure.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Referer form structure
+
+Once you visit `/start` enter your email address and the OTP sent to your email you'll be sent to the eligibility screener.
```
Comment (vassyz) [inline docs/referrer-form-structure.md:3]: I think this needs better wording.

File: docs/referrer-form-structure.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+This will filter out those applications who are ineligible for this form.
```
Comment (vassyz) [inline docs/referrer-form-structure.md:5]: Maybe try to say that the applicants only have access to their own forms or something related to authentication and authorisation. Filtering out ineligible applications sounds like something else to me but maybe we're taking about two different things.

File: docs/referrer-form-structure.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+Until we show them all the steps and confirm here
+
+```
+Referrals::ReferrersController#show
+```
+
+### Form objects
+
+Each of the steps above has it's own form object so for instance [app/controllers/referrals/referrer_phone_controller.rb](../app/controllers/referrals/referrer_phone_controller.rb) uses [app/forms/referrer_phone_form.rb](../app/forms/referrer_phone_form.rb)
```
Comment (vassyz) [inline docs/referrer-form-structure.md:58]: Minor typo `its own`.

File: docs/referrer-form-structure.md (hunk 1 above)
Comment (gpeng) [reply docs/referrer-form-structure.md:3]: There's a ticket to change this flow too that we'll do this sprint so this is going to change in any case. We could probably leave this whole bit out and just have the eligibility check start with the start button. The email signup is going to be moved to the end of the process as per the prototype.

Comment (paulodeon) [reply docs/referrer-form-structure.md]: I've addressed that now

**Review by vassyz — APPROVED** [review]

**Pull request closed without merging**